modules/deployment-model/src/test/java/org/mule/runtime/deployment/model/internal/domain/DomainClassLoaderFactoryTestCase.java

```diff
@@ -95,6 +96,33 @@ public void createClassLoaderFromDomainDescriptor() {
     assertThat(domainClassLoader.getArtifactId(), equalTo(artifactId));
   }
 
+  @Test
+  public void secondClassLoaderCreationReturnsCachedInstance() {
```

Comment:
merge these two tests in one verify that before disposing the same instance is returned from the factory and after disposing you get a new instance.